src/dex/uniswap-v2/uniswap-v2.ts

```diff
@@ -105,6 +107,8 @@ export class UniswapV2EventPool extends StatefulEventSubscriber<UniswapV2PoolSta
         ' pool',
       logger,
     );
+
+    this.iface = new Interface(decoderABI);
```

Comment:
This is actually very inefficient, you are constructing a new Interface for every pool.  You could pass in the `Interface` directly instead of `JsonFragment[]` and creating it every time in the constructor.